### DISCUSSION TITLE
`C-S-;` and `A-S-;` keybindings don't do anything in any mode

### DISCUSSION THREAD

**Discussion by dementive** [helix-editor/helix#10844] [General]
### Summary

Im trying to setup my keybindings in helix like I have in my other editor and have most of it working except extending the selection to the end of the line with these keybindings:

```
# Move beginning/end of line and select
"C-S-h" = "extend_to_line_start"
"C-S-;" = "extend_to_line_end"
"A-S-h" = "extend_to_line_start"
"A-S-;" = "extend_to_line_end"
```

"C-S-h" works no problem and so does extend_to_line_end but it seems like "C-S-;". It seems like the problem might be it is ignoring the keybinding because it used Shift and the ; but idk.

### Reproduction Steps

I tried this:

Adding these 2 lines to the config.toml for any mode
```
"C-S-;" = "extend_to_line_end"
"A-S-;" = "extend_to_line_end"
```

I also tried 
```
"C-S-:" = "extend_to_line_end"
"A-S-:" = "extend_to_line_end"
```
with the same results.

I expected this to happen:

It should extend the selection to the end of the line, it works perfectly fine with other keybindings but this specific one does not work for some reason.

Instead, this happened:

In insert mode it just places a `:` in the buffer and in the other modes it just does nothing, it seems like helix just doesn't load the keybinding at all and there are no config errors with it either.

### Platform

LInux

### Terminal Emulator

Alacritty

### Installation Method

Pacman

### Helix Version

24.03

**Comment by pascalkuthe** [comment]
using shift in terminal keybindings doesn't really work. We normalize `C-S-h` to `C-H` internally but the same can not be done for modifiers (since its generally locale specific). Terminals usually don't emit shift modifiers directly.

So there will probably never be an `a-s-;` event emitted by a terminal emulator (depends on os, locale and many other things tough). Generally: directly specify whatever character you could get in insertmode when pressing shift and the key you want to map. Using shift as a modifier is brittle due to the way terminals handle inputs
